### PR TITLE
[WIP] Issue #1935 Implement ability to run minishift against an existing VM

### LINKED ISSUE #1935: Spike : Implement ability to run Minishift against an existing VM
Refer: https://github.com/minishift/minishift/issues/392

### PR DESCRIPTION
**Pull request by praveenkumar**
Fix #1935 
@LalatenduMohanty  This is the POC code which I did long back. I rebased and modified according to driver codebase. Please have a look and carry forward on it or change according to your understanding.

### CODE REVIEW

**Comment by praveenkumar** [comment]
To test this PR you need to follow:

On the remote host:

```
1. Create a user which have sudo permission without the password.
2. Make sure firewall allowed the 2376 port once docker connection is established then this port is used for docker daemon communication.
```

Command to run.
```
./minishift start --vm-driver generic --remote-ip-address 10.70.48.27 --remote-ssh-username <username> --show-libmachine-logs --v=5
```

**Comment by gbraad** [comment]
> Create a user which have sudo permission without the password.

Why without a password? You mean, password-less, by using certificates... right (`ssh-copy-id`) ?

**Comment by praveenkumar** [comment]
> Why without a password? You mean, password-less, by using certificates... right (ssh-copy-id) ?

@gbraad right, passwordless sudoer. The same user also going to provision the VM like install docker if not installed add systemd dropin file add certificates, restart docker daemon ..etc.

File: pkg/minikube/constants/constants_linux.go [hunk 1]
```diff
@@ -21,6 +21,7 @@ package constants
 var SupportedVMDrivers = [...]string{
 	"virtualbox",
 	"kvm",
+	"generic",
```
Comment (gbraad) [inline pkg/minikube/constants/constants_linux.go]: calling this maybe `remote` makes more sense

**Comment by LalatenduMohanty** [comment]
Closing the PR as I will create another one with modified code.

**Pull request closed without merging**